packages/core/src/html/UlNodeProcessor.ts

```diff
@@ -0,0 +1,265 @@
+import cheerio from 'cheerio';
+import get from 'lodash/get';
+import { MbNode, NodeOrText } from '../utils/node';
+
+const { processIconString } = require('../lib/markdown-it/plugins/markdown-it-icons');
+const emojiDictionary = require('../lib/markdown-it/patches/markdown-it-emoji-fixed');
+
+interface EmojiData {
+  [key: string]: string;
+}
+
+const emojiData = emojiDictionary as unknown as EmojiData;
+
+const ICON_ATTRIBUTES = ['icon', 'i-width', 'i-height', 'i-size', 'i-class'];
+
+interface IconAttributes {
+  icon?: string;
+  className?: string;
+  size?: string;
+  width?: string;
+  height?: string;
+}
+
+function classifyIcon(icon: string) {
+  const isEmoji = Object.prototype.hasOwnProperty.call(emojiData, icon);
+  const localFileRegex = new RegExp([
+    '^(./)?',
+    '[a-zA-Z0-9-._~:/?#[@!$&\'()*+,;=%]+',
+    '.(jpg|png|gif|bmp|svg|jpeg|webp|avif)$',
+  ].join(''));
+
+  const urlRegex = /^(http(s)?|ftp):\/\/[^\s/$.?#].[^\s]*$/;
+  const isImage = !isEmoji && (localFileRegex.test(icon) || urlRegex.test(icon));
+  return {
+    isEmoji,
+    isImage,
+    unicodeEmoji: isEmoji
+      ? emojiData[icon]
+      : undefined,
+  };
+}
+
+function createIconSpan(iconAttrs: IconAttributes): cheerio.Cheerio {
+  const {
+    isEmoji,
+    isImage,
+    unicodeEmoji,
+  } = classifyIcon(iconAttrs.icon!);
+
+  let spanContent;
+
+  if (isEmoji) {
+    spanContent = `<span aria-hidden="true">${unicodeEmoji}</span>`;
+  } else if (isImage) {
+    const img = cheerio(`<img src='${iconAttrs.icon}' alt='Icon'>`)
+      .css({ width: iconAttrs.width, height: iconAttrs.height, display: 'inline-block' })
+      .addClass(iconAttrs.className || '');
+    img.append('\u200B');
+
+    spanContent = cheerio('<span></span>').append(img).css({
+      'padding-bottom': '0.3em',
+      'padding-top': '0.3em',
+    });
+  } else {
+    spanContent = processIconString(iconAttrs.icon);
+  }
+
+  let spanNode = cheerio(spanContent).css({ 'font-size': 'unset', 'min-width': '16px' });
```

Comment:
Should `min-width` be set when using icons? What if the icon's width `< 16px`? (extra spacing could look strange)

Reply:
@SPWwj sorry, I actually think having this for icons is a good thing to ensure alignment for common cases. (if it's the intention)

The first question should have been:

> Should min-width *only* be set when using icons?

(should images have this?)

Reply:
> What if the icon's width < 16px? (extra spacing could look strange)

This is a separate thought on icons (if there are icons that are really small, e.g. 8px). Are there such cases?

Reply:
Another unrelated thing: we can define this as a factor of `em` instead of `px` as well since icons scale according to font size.

Reply:
> (should images have this?)
I don't believe the `min-width: 16px` is necessary for images or icons, since images have a hard dimension set by width and height. 

Originally, I introduced `min-width: 16px` to standardize and align the default size for all icons (except images), as different frameworks (e.g., FontAwesome and Material) have different font sizes. Setting a minimum width could help to align these different font sizes, at least for the default font size. 

> This is a separate thought on icons (if there are icons that are really small, e.g. 8px). Are there such cases?

However, as you've pointed out, there could be cases where users might want to replace the default unordered list bullet with a customized icon, which could potentially be really small, less than 8 px. 

Moreover, as we previously discussed, we provide users the freedom to set the dimensions of the icon. Therefore, the alignment issue mentioned above can be easily overcome by adjusting the font size if a user chooses to mix icons from different frameworks. 

Based on the two points above, I believe there is no need to include `min-width`.

Reply:
> easily overcome

I'm not sure it is easy, these are not monospace "fonts".

Using your example,
![image](https://github.com/MarkBind/markbind/assets/3306138/d902a4f8-5931-417b-b3eb-154cb2316931)

This is what I had to edit things to, to make things "work":

![image](https://github.com/MarkBind/markbind/assets/3306138/7105c6fa-a857-4f23-b3f7-60ac419f3670)

It is not optimal as well, as the height scales accordingly (e.g. the pull request icon looks overly large).

Reply:
Ideally, we have [this](https://github.com/MarkBind/markbind/pull/2316#discussion_r1252113871):

> Another thing if you have the time, you can consider is providing an alignment option for the case that icon widths are different (e.g. images of different sizes); But I think this is rarer and less important for now, we can pick a good default and add it later if needed.

i.e. think of icons on the same level as existing in a single column direciton flex, then we support things like `align-self: start, center, stretch, end, etc.` more intuitively.

But as its going to require a fair bit of work on top of this PR, 16px seems like a good compromise for now.